### ISSUE TITLE
RTRIM not working as expected

### ISSUE THREAD

**Issue by deiks** [arangodb/arangodb#9558]
## My Environment

* __ArangoDB Version__:        3.4.7
* __Storage Engine__:             
* __Deployment Mode__:       
* __Deployment Strategy__:   
* __Configuration__:               
* __Infrastructure__:               
* __Operating System__:        Ubuntu 14.04.5 LTS 
* __Total RAM in your machine__:        
* __Disks in use__:        
* __Used Package__:              

## Component, Query & Data

__Affected feature__:

__AQL query (if applicable)__: RTRIM("20d", "d")

__AQL explain (if applicable)__:

__Dataset__:

__Size of your Dataset on disk__:

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

1. Create simple query with **RETURN RTRIM("20d", "d")**
2. 
3. 

__Problem__:

Not working as expected. Returns same value.

__Expected result__:

Result "20"

Works fine on 3.3.12

**Comment by jsteemann** [comment]
Just ran the following code in my 3.4.7 arangosh against a 3.4.7 single server:
```
> db._query('RETURN RTRIM("20d", "d")');
[object ArangoQueryCursor, count: 1, cached: false, hasMore: false]

[ 
  "20" 
]
```
This seems to be the correct result.
Can you let us know how to produce a wrong result? Is there anything special about your setup?

**Comment by deiks** [comment]
I tried via Web interface. Nothing special, just type line in query editor.

It's vagrant machine with older version of Ubuntu as described. Maybe that could be a reason.

**Comment by deiks** [comment]
Strange thing. After new test, seems like works fine when using **20d**, but if you sed **2d** then it fails.

**Comment by jsteemann** [comment]
It turned out that `RTRIM("2d", "d")` does not work as expected. `RTRIM("20d", "d")` seems to work fine.
A bugfix for the issue can be found in the PRs referenced above.
The bugfixes will be released with 3.4.8 and 3.5.x (version yet unknown).

**Comment by maxkernbach** [comment]
Closing this issue since versions 3.4.8 and 3.5.0, which include a fix for `RTRIM()`, have been released.
`RTRIM("2d", "d")` now returns the correct result.

Please find the latest version in the [download section](https://www.arangodb.com/download-major/).
